android/src/main/java/net/no_mad/tts/TextToSpeechModule.java

```diff
@@ -580,10 +599,11 @@ private int stop() {
 
     @ReactMethod
     public void stop(Promise promise) {
-        if (notReady(promise)) return;
-
-        int result = stop();
-        boolean resultValue = (result == TextToSpeech.SUCCESS) ? Boolean.TRUE : Boolean.FALSE;
+        boolean resultValue = Boolean.TRUE;
+        if (ready != null || ready == Boolean.TRUE) {
```

Comment:
Do you mean `(ready != null && ready == Boolean.TRUE)` ? This condition will pass when `ready == false`

Reply:
Yes should be && not || (difficult to get the false case in anyway that is why it passed my testing)

Reply:
fixed with commit: [d324968](https://github.com/tomtom-international/react-native-tts/pull/2/commits/d324968c7354148a8f7405b9376005661508e879)